backend/lm_backend/api/routes/configurations.py

```diff
@@ -24,11 +34,54 @@
     status_code=status.HTTP_201_CREATED,
 )
 async def create_configuration(
-    configuration: ConfigurationCreateSchema,
+    configuration: ConfigurationCompleteCreateSchema,
     secure_session: SecureSession = Depends(secure_session(Permissions.CONFIG_EDIT)),
 ):
-    """Create a new configuration."""
-    return await crud.create(db_session=secure_session.session, obj=configuration)
+    """
+    Create a new configuration.
+
+    The features for the configuration will be specified in the request body.
+    If the feature's product doesn't exist, it will be created.
+    """
+    configuration_created: Configuration = await crud_configuration.create(
+        db_session=secure_session.session,
+        obj=ConfigurationCreateSchema(**configuration.dict(exclude={"features", "license_servers"})),
+    )
+
+    if configuration.features:
+        try:
+            for feature in configuration.features:
+                feature_obj = {
+                    "name": feature.name,
+                    "product_id": feature.product_id,
+                    "config_id": configuration_created.id,
+                    "reserved": feature.reserved,
+                }
+                await crud_feature.create(
+                    db_session=secure_session.session, obj=FeatureCreateSchema(**feature_obj)
+                )
+        except HTTPException:
+            await crud_configuration.delete(db_session=secure_session.session, id=configuration_created.id)
```

Comment:
I don't think you need to explicitly delete. If an exception is raised, the whole transaction should be rolled back including any rows created.

Reply:
I removed the delete but the configuration row is still there after the failure case.
It looks like the creation is being done in different transactions.